### PR TITLE
Feature/issue 48 add optional cas login

### PR DESCRIPTION
**Pull request by nathanbegbie**

### CODE REVIEW

**Comment by nathanbegbie** [comment]
@miltontony Please review

File: molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/backends.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from django_cas_ng.backends import CASBackend
+
+
+class MoloCASBackend(CASBackend):
+
+    def authenticate(self, ticket, service, request):
+        user = super(
+            MoloCASBackend, self).authenticate(ticket, service, request)
+        if user is None:
+            print 'ticket is not valid'
```
Comment (miltontony) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/backends.py]: stray print

File: molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/backends.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            return None
+
+        print request.session['attributes']
```
Comment (miltontony) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/backends.py]: printing

File: molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/urls.py [hunk 3]
```diff
@@ -10,25 +10,42 @@
...
+    )
+# implement CAS URLs in a production setting
+else:
+    urlpatterns = patterns(
+        '',
+        url(r'^django-admin/', include(admin.site.urls)),
+        url(r'^admin/login/', 'django_cas_ng.views.login'),
+        url(r'^admin/logout/', 'django_cas_ng.views.logout'),
+        url(r'^admin/callback/', 'django_cas_ng.views.callback'),
+        url(r'^admin/', include(wagtailadmin_urls)),
```
Comment (miltontony) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/urls.py]: Common url patterns can be refactored outside this if/else block using `urlpatterns += (..)

**Comment by miltontony** [comment]
`backends.py` and `middleware.py` should live in `molo/core/`

**Comment by nathanbegbie** [comment]
@miltontony: decrease in coverage, but I'm not sure how we can test this. Thoughts?

File: molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/urls.py (hunk 3 above)
Comment (nathanbegbie) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/urls.py]: This breaks for some reason and gives a 404 for `/admin/` when using dev settings

**Comment by miltontony** [comment]
@nathanbegbie ready for your review

**Comment by codiebeulaine** [comment]
@miltontony ready for review

**Pull request merged**